src/main/java/com/minecolonies/coremod/compatibility/jei/transer/PrivateCraftingTeachingTransferHandler.java

```diff
@@ -39,26 +42,34 @@
     @Nullable
     @Override
     public IRecipeTransferError transferRecipe(
-      final CraftingGUIBuilding craftingGUIBuilding, final IRecipeLayout recipeLayout, final EntityPlayer entityPlayer, final boolean b, final boolean b1)
+            final CraftingGUIBuilding craftingGUIBuilding,
+            final IRecipeLayout recipeLayout,
+            final EntityPlayer entityPlayer,
+            final boolean b,
+            final boolean b1)
     {
-        IGuiItemStackGroup itemStackGroup = recipeLayout.getItemStacks();
+        final IGuiItemStackGroup itemStackGroup = recipeLayout.getItemStacks();
 
         // indexes that do not fit into the player crafting grid
-        Set<Integer> badIndexes = ImmutableSet.of(2, 5, 6, 7, 8);
+        final Set<Integer> badIndexes = ImmutableSet.of(2, 5, 6, 7, 8);
 
         // compact the crafting grid into a 2x2 area
-        Map<Integer, ItemStack> guiIngredients = new HashMap<>();
+        final Map<Integer, ItemStack> guiIngredients = new HashMap<>();
         guiIngredients.put(0, ItemStackUtils.EMPTY);
         guiIngredients.put(1, ItemStackUtils.EMPTY);
         guiIngredients.put(3, ItemStackUtils.EMPTY);
         guiIngredients.put(4, ItemStackUtils.EMPTY);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)